### ISSUE TITLE
Call to undefined function Webklex\IMAP\imap_open() - Mac OS

### ISSUE THREAD

**Issue by FounderCasey** [Webklex/laravel-imap#279]
Hello, 

I have a Laravel 6 and PHP 7.4.1 project looking to incorporate Laravel IMAP. 

I had some trouble installing the package, but finally got composer to install it. Now I am getting this error when trying to connect.

`Call to undefined function Webklex\IMAP\imap_open()
`

I have read that this could be an issue with PHP not finding my IMAP extension.

When I run php -i | grep imap

**Protocols**:
`
Protocols => dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtmp, rtsp, scp, sftp, smb, smbs, smtp, smtps, telnet, tftp
`

**Full return**:
`Additional .ini files parsed => /usr/local/etc/php/7.4/conf.d/ext-imap.ini,
Protocols => dict, file, ftp, ftps, gopher, http, https, imap, imaps, ldap, ldaps, pop3, pop3s, rtmp, rtsp, scp, sftp, smb, smbs, smtp, smtps, telnet, tftp
imap
imap.enable_insecure_rsh => Off => Off`

Any pointers?

**Comment by FounderCasey** [comment]
@Webklex Sorry, not sure I understand the reference... New to php and I am unable to find any info. Is there something I should change?

**Comment by Webklex** [comment]
Hi @FounderCasey ,
please give the current master version a try :)

**Comment by FounderCasey** [comment]
@Webklex I went ahead and tried that but the error just changed to 

`Call to undefined function imap_open()` instead of `Call to undefined function Webklex\IMAP\imap_open()`

**Comment by Webklex** [comment]
Please check the output of php_info(). The php-imap module might not be enabled on your Webserver? Yes I know your cli does say its installed but your apache or nginx has its own config (php.ini).

**Comment by FounderCasey** [comment]
`user "MYUSER" staff;
worker_processes auto;

events {
    worker_connections  1024;
}

http {
    include mime.types;
    default_type  application/octet-stream;

    sendfile on;
    keepalive_timeout  65;
    types_hash_max_size 2048;
    client_max_body_size 128M;

    server_names_hash_bucket_size 128;

    gzip  on;
    gzip_comp_level 5;
    gzip_min_length 256;
    gzip_proxied any;
    gzip_vary on;
    
    ssi on;

    gzip_types
    application/atom+xml
    application/javascript
    application/json
    application/rss+xml
    application/vnd.ms-fontobject
    application/x-font-ttf
    application/x-web-app-manifest+json
    application/xhtml+xml
    application/xml
    font/opentype
    image/svg+xml
    image/x-icon
    text/css
    text/plain
    text/x-component;

    include "/Users/MYPATH/.config/valet/Nginx/*";
    include servers/*;
    include valet/valet.conf;
}
`

This is all I could find in regards to an actual config file for Nginx... This must be my set up.

**Comment by Webklex** [comment]
I did some more research.
I've never used nginx but it seems like you have to use php-fpm as an intermediate proxy to use any php module.
```bash
sudo apt-get install php5-imap
sudo php5enmod imap 
service php5-fpm restart
sudo service nginx restart
```

**Comment by FounderCasey** [comment]
Still nothing... Well, this sucks. Been a week now and I have yet to find a fix.

Thank you for your responsiveness and effort to help!

**Comment by FounderCasey** [comment]
@Webklex I factory reset my workspace and I found out that all the requirements except for php-imap come in `brew install php`, the latest version(7.4.2). First try, everything works great. 

I am not sure where I went wrong, I believe it was because I had different .ini files and I had no clue where they were. Lots of random errors here and there. 

Again, thank you for your help and this package!
